dapp_user/application/handlers/user_handlers.py

```diff
@@ -0,0 +1,26 @@
+import json
+
+from common.constant import StatusCode
+from common.exceptions import BadRequestException
+from common.logger import get_logger
+from common.utils import validate_dict, handle_exception_with_slack_notification, generate_lambda_response
+from dapp_user.config import SLACK_HOOK, NETWORK_ID
+from dapp_user.domain.services.user_service import UserService
+
+logger = get_logger(__name__)
+
+
+@handle_exception_with_slack_notification(SLACK_HOOK=SLACK_HOOK, NETWORK_ID=NETWORK_ID, logger=logger)
+def add_or_update_user_preference(event, context):
+    payload = json.loads(event["body"])
+    username = event["requestContext"]["authorizer"]["claims"]["email"]
+    required_keys = ["communication_type", "preference_type", "source", "status"]
+    if not validate_dict(payload, required_keys):
+        raise BadRequestException()
```

Comment:
every service will have their own exception,can we move this exception to user servcie